File: 백준/Bronze/2884. 알람 시계/알람 시계.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.on("line", (v) => {
  arr = v.split(" ").map((v) => +v);

  if (arr[1] >= 45) {
    console.log(`${arr[0]} ${arr[1] - 45}`);
  } else if (arr[0] === 0) {
    console.log(`23 ${arr[1] + 15}`);
  } else {
    console.log(`${arr[0] - 1} ${arr[1] + 15}`);
  }
  rl.close();
});

rl.on("close", () => {
  process.exit();
});

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
A, B = input().split()
A = int(A)
B = int(B)
if(B>=45):
    print(str(A)+" "+str(B-45))
elif(A==0):
    print("23 "+str(B+60-45))
else:
    print(str(A-1)+" "+str(B+60-45))